trillian/ctfe/dns_handlers.go

```diff
@@ -0,0 +1,324 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// These handlers implement the CT over DNS proposal documented at:
+// https://github.com/google/certificate-transparency-rfcs/blob/master/dns/draft-ct-over-dns.md
+
+package ctfe
+
+import (
+	"context"
+	"encoding/base32"
+	"encoding/base64"
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/golang/glog"
+	ct "github.com/google/certificate-transparency-go"
+	"github.com/google/certificate-transparency-go/tls"
+	"github.com/google/certificate-transparency-go/trillian/ctfe/configpb"
+	"github.com/google/trillian"
+	"github.com/miekg/dns"
+)
+
+// The maximum length of the text returned in a DNS consistency or inclusion
+// proof response.
+const maxProofLen = 255
+
+// The client omits the padding from the hash query so we need to add it back.
+const base32LeafPad = "==="
+
+const (
+	// Used for STH responses, which change over time.
+	sthTTL = 5 * 60
```

Comment:
use time.Durations here?

Reply:
Could do but the DNS API just takes ints.